packages/mui-base/src/BadgeUnstyled/BadgeUnstyled.js

```diff
@@ -91,33 +91,21 @@ const BadgeUnstyled = React.forwardRef(function BadgeUnstyled(props, ref) {
 
   const classes = useUtilityClasses(ownerState);
 
-  const Root = components.Root || component;
-  const rootProps = componentsProps.root || {};
+  const Root = component || components.Root || 'span';
```

Comment:
I have probably missed a train 🚋 here. I was expecting that developers could do:

```jsx
// in packages/my-design-system/Badge.tsx
import { BadgeUnstyled } from '@mui/base';

const components = {
  Root: styled('span')({
    backgroundColor: 'red',
  }),
  Badge: styled('span')({
    backgroundColor: 'yellow',
  }),
};

export default function MyBadge(props) {
  return <BadgeUnstyled components={components} {...prop};
}

// in packages/web/signin.tsx
import { MyBadge } from 'packages/my-design-system';

export default function MyScreen() {
  return <div><MyBadge component="div" /></div>
}
```

The previous behavior seemed better. Why? Because `packages/mui-material/src/Badge/Badge.js` now as more LOCs. I believe the purpose of the base components is to own the complexity so the styled version can be as simple as possible.

Thoughts?